internal/server/landing_page.go

```diff
@@ -247,70 +248,40 @@ func fetchBtData(
 		return nil, err
 	}
 	// Populate result from landing page cache
-	result := map[string]map[string]*ObsTimeSeries{}
+	result := map[string]*pb.StatVarSeries{}
 	for dcid, data := range baseDataMap {
-		landingPageData := data.(*LandingPageData)
-		finalData := map[string]*ObsTimeSeries{}
+		landingPageData := data.(*pb.StatVarObsSeries)
+		finalData := &pb.StatVarSeries{Data: map[string]*pb.Series{}}
 		for statVarDcid, obsTimeSeries := range landingPageData.Data {
-			obsTimeSeries.filterAndRank(&ObsProp{})
-			finalData[statVarDcid] = obsTimeSeries
+			finalData.Data[statVarDcid] = getBestSeries(obsTimeSeries)
 		}
 		result[dcid] = finalData
 	}
 
-	// The landing page cache may depends on out dated chart config. Fetch stats
-	// that are not included in the landing page cache.
-	gotStatVars := map[string]struct{}{}
-	for statVar := range result[places[0]] {
-		gotStatVars[statVar] = struct{}{}
-	}
-	missingStatVars := []string{}
-	for _, statVar := range statVars {
-		if _, ok := gotStatVars[statVar]; !ok {
-			missingStatVars = append(missingStatVars, statVar)
-		}
-	}
-	if len(missingStatVars) > 0 {
-		type svData struct {
-			sv   string
-			data map[string]*ObsTimeSeries
-		}
-		errs, errCtx := errgroup.WithContext(ctx)
-		dataChan := make(chan svData, len(missingStatVars))
-		for _, statVar := range missingStatVars {
-			statVar := statVar
-			errs.Go(func() error {
-				resp, err := s.GetStats(errCtx, &pb.GetStatsRequest{
-					Place:    places,
-					StatsVar: statVar,
-				})
-				if err != nil {
-					return err
-				}
-				var tmp map[string]*ObsTimeSeries
-				err = json.Unmarshal([]byte(resp.Payload), &tmp)
-				if err != nil {
-					return err
-				}
-				dataChan <- svData{sv: statVar, data: tmp}
-				return nil
-			})
-		}
-		err := errs.Wait()
+	// Fetch addtioinal stats as requested.
+	if len(statVars) > 0 {
+		resp, err := s.GetStatSetSeries(ctx, &pb.GetStatSetSeriesRequest{
```

Comment:
For my understanding, does the website then change to stop setting the extra stat-vars in prod?

https://github.com/datacommonsorg/website/blob/175e9c62f64e62b89f8a1bcc2ed4bfb599c7c6a0/server/routes/api/landing_page.py#L382

Reply:
along those lines, since this is a website specific api change that isn't backwards compatible, should this change be made in the website repo, along with corresponding flask changes? we might not be able to autopush otherwise.